lib/bamboo/attachment.ex

```diff
@@ -2,24 +2,36 @@ defmodule Bamboo.Attachment do
   @moduledoc """
   """
 
-  defstruct filename: nil, content_type: nil, path: nil, data: nil
+  defstruct filename: nil, content_type: nil, path: nil, data: nil, content_id: nil
 
   @type t :: %__MODULE__{
           path: nil | String.t(),
           filename: nil | String.t(),
           content_type: nil | String.t(),
-          data: nil | binary()
+          data: nil | binary(),
+          content_id: nil | String.t()
         }
 
   @doc ~S"""
   Creates a new Attachment
 
+  context_id can be use to embed an image, attach it and reference in the message body by
+  setting its CID (Content-ID) and using a standard HTML tag:
+
+    <img src="cid:some-image-cid" alt="img" />
+
+  within a HTML email message.
+
   Examples:
 
       Bamboo.Attachment.new("/path/to/attachment.png")
       Bamboo.Attachment.new("/path/to/attachment.png", filename: "image.png")
-      Bamboo.Attachment.new("/path/to/attachment.png", filename: "image.png", content_type: "image/png")
+      Bamboo.Attachment.new("/path/to/attachment.png", filename: "image.png", content_type: "image/png", content_id: "12387432")
       Bamboo.Attachment.new(params["file"]) # Where params["file"] is a %Plug.Upload
+
+      email
+        |> put_html_layout({LayoutView, "email.html"})
+        |> put_attachment(%Bamboo.Attachment{content_type: "image/png", filename: "logo.png", data: get_content(), content_id: "2343333333"})
```

Comment:
The code formatter would line the pipes up directly with `email` - so all the lines are directly in line. Also, what do you think about switching the `data` passed in in the example to some arbitrary data instead of referencing a `get_content()` function, to be as understandable as possible?